\raggedright

# Taxon `r taxon`

## 1. Query source chronograms
There are `r length(tax_dq$cleaned_names)` species in the taxon `r taxon`, following the NCBI taxonomy database.
Information on time of divergence is available for
```{r echo = FALSE, results = 'asis'}
tt <- nrow(tax_median_phylo$taxon_distribution)
if(tt == length(tax_dq$cleaned_names)){
  cat("all")
} else {
  cat(nrow(tax_median_phylo$taxon_distribution))
}
```
of these species across `r length(tax_dr)` published and peer-reviewed chronograms.
Original study citations as well as number of `r taxon` species found across those source
chronograms is shown in `r table_chronograms`. `r ultrametricity` and their maximum
ages range from `r round(min(tax_summary$mrca), digits = 3)` to
`r round(max(tax_summary$mrca), digits = 3)` million years ago (MYA).
As a means for comparison, lineage through time plots of all source chronograms
available in data base are shown in figure \ref{fig:lttplot_phyloall}.


## 2. Summarize results from query

LTT plots are a nice way to visually compare several trees. But what if you want
to summarize information from all source chronograms into a single summary chronogram?

The first step is to identify the degree of species overlap among your source chronograms: if each
source chronogram has a unique sample of species, it will not be possible to combine
them into a single summary chronogram. To identify the set of trees or _grove_ with the most source
chronograms that have at least two overlapping taxa, we followed AnÃ© et al. 2016.
```{r echo = FALSE, warning=FALSE, message = FALSE, results = 'asis'}
if(length(tax_bestgrove) == length(tax_dr)){
    cat("In the case of the ", taxon, " all source chronograms have at least two overlapping species.\n")
    cat("\nNow that we know that the best grove has all source chronograms ")
} else {
    cat("In this case, not all source chronograms found for the ", taxon,
        " have at least two overlapping species. The largest grove has ",
            length(tax_bestgrove), " chronograms (out of ", length(tax_dr),
            " total source chronograms).\n")
    cat("\nNow that we have identified a grove ")
}
```
we can go on to summarize it by translating the source chronograms into patristic distance matrices and
then averaging them into a single summary matrix; yes, this first step is _that_
straightforward. We can average the source matrices by simply using the mean or
median distances, or we can use methods that involve transforming
the original distance matrices --such as the super distance matrix (SDM) approach of Criscuolo et al. 2006-- by minimizing
the distances across source matrices. As a result of such transformation, an SDM
summary matrix can contain negative values. `r negs_sdm`

Because our summary matrix is basically a distance matrix, a distance-based clustering
algorithm could be used to reconstruct the tree. Algorithms such as neighbour joining (NJ) and
unweighted pair group method with arithmetic mean (UPGMA) are
fast and work very well when there are no missing values in the matrices. However, summary
matrices coming from source chronograms usually have several NAs and missing rows.
<!--This data set for example has NUMBER cells with missing data.-->
When this happens, variants of traditional clustering algorithms have been developed to deal with missing values.
<!--
NJS: ape::njs
UPGMA: daisy + upgma
BIONJ: ape::bionjs(X)
Minimum Variance Reduction: ape::mvrs(X, V)
Triangle Method, Tree reconstruction based on the triangle method: ape::triangMtds(X)
-->
However, even these methods do not work well with our summary matrices, as shown in the following section.
We should note that these clustering methods are usually applied to distance matrices representing substitution rates
and not absolute time.

###   2.1. Clustering a summary matrix

NJ, UPGMA, BIONJ, minimum variance reduction (MVR) and the triangle method (TM)
algorithms were used to cluster median and SDM summary distance matrices.
None of these clustering algorithms returned trees matching source chronograms
(Fig. \ref{fig:lttplot_cluster_median}, Appendix Fig. \ref{fig:lttplot_cluster_sdm}).
UPGMA is the only algorithm that returns ultrametric trees, but they are considerably
older than expected from ages observed in source chronograms.
The other methods returned trees with ages that coincide with those observed in
source chronograms. However, they resulting chronograms are not ultrametric.
To overcome the issues presented by clustering algorithms, we used all data avilable in the summary
matrix as calibrations over a consensus tree to obtain a summary chornogram.


###   2.2. Calibrating a consensus tree with data from a summary matrix

Even if the branch lengths coming form the clustered chronograms are not adequate,
the topology can still be used as a backbone tree that can be dated using data from
the summary matrix as secondry calibrations.
A summary of divergence times available for each node can be obtained from the summary matrix,
simply by getting the nodes from the backbone tree that correspond to each pair of
taxa in the matrix. Finally, this summary of node divergence times can be used with
the consensus tree as input in any dating software that does not require data.
The branch length aduster (BLADJ) algorithm [@Webb2000] is really fast and does
not make any evolutionary assumptions on age distribution. Other software such as
MrBayes and r8s can be used instead of BLADJ by running them without data.
In here, we show summary chronograms obtained using minimum, mean and maximum distances
from the summary of node divergence times of the backbone tree as fixed ages in
BLADJ (`r fig_lttplot_median_and_sdm`).
Summary chronograms from both types of summary matrices are quite similar. As expected,
SDM chronograms using minimum, mean and maximum distances do not vary much in their
maximum age, because ages are transformed to minimize the variance. In contrast,
the median chronograms obtained with minimum, mean and maximum distances have wider variation
in their maximum ages, as can be observed in the distance between the green arrows
in `r fig_lttplot_median_and_sdm`. This variation simply represents variation in source data.


## 3. Generate new chronograms

Another way to leverage information from the source chronograms is to use their node
ages as secondary calibration points to date any tree topology (with or without
branch lengths) given that at least two taxa from source chronograms are in
the tips of that topology.
In this data set we have `r nrow(tax_allcal)` calibrations in total (that basically
corresponds to the sum of the number of nodes from each source chronogram).
Once we have a target tree topology, we can map the calibrations to the target tree.
Some nodes will have several calibrations and some others might have none. Also,
some node ages can be conflicting, with descendant nodes being older than parent nodes.
We performed a series of cross validation analyses with different dating methods, by
dating the topologies of each source chronogram using information from all other
source chronograms as calibration points.

### 3.1. Calibrate a tree without branch length data
To date a tree in the absence of data on relative evolutionary rates (molecular
or morphological) we follow the same methodology as the one used to obtain summary
chronograms. First, we obtained the nodes that correspond to each pair of taxa in the data
set of total calibrations to construct a summary of node calibrations for the backbone
tree. Then, we used mean ages as secondary calibrations for the backbone tree with
the software BLADJ. In general, the time of divergence information from other source
chronograms allows to recover the divergence times from the original study. In some cases,
it is evident that information from a particular study really affects the summary
of divergence times. In some other cases, the root of the tree is not calibrated.
Since BLADJ has no underlying model of evolution, there is no way for the algorithm
to calculate this age. To fix this, we simply added a unit of the mean difference across
ranked ages from secondary calibrations (`r fig_lttplot_bladj`).

### 3.2. Calibrate a tree with data
If you have a tree with branch lengths proportional to relative substitution rates,
you can use the source chronogram node ages as secondary calibrations with various
algorithms for phylogenetic dating to get branch lengths proportional to absolute time.
To exemplify this, we got DNA markers from the Barcode of Life Database (BOLD)
to estimate branch lengths as relative DNA substitution rates on a tree topology of our choosing.
In this example we retrieved data from the cytochrome C oxidase subunit I (COI) marker, that is of
widespread use in barcoding, providing DNA data for a very wide number of organisms.
<!-- source_chronogram_bold_tree <- make_bold_otol_tree(input = source_chronogram_topology,
marker = "COI", otol_version = "v3", chronogram = TRUE)
source_chronogram_bold_tree_notc <- make_bold_otol_tree(input = source_chronogram_topology,
marker = "COI", otol_version = "v3", chronogram = FALSE)
-->
```{r echo = FALSE, warning=FALSE, message = FALSE, results = 'asis'}
# make_bold_otol_tree always returns a tree, even if it fails (it will return teh original tree or the otol tree if it constructed one)
# so to know if it is succesfull, check if the output has branch lengths
if(all(bold_has_brlen(tax_phyloall_boldi))){
    paragraph <- paste0("Unfortunately, a tree with branch lengths could not be
    constructed for any of the source chronograms available for the ", taxon,
    ", so this workflow will not be exemplified here. This can happen for several
    reasons. If the tree has only two tips, the tree search cannot be performed. If the Please look into other DateLife
    examples available in here for more infromation about this workflow.")
} else {
    paragraph <- paste0("A tree with branch lengths could be constructed for ",
        length(bold_has_brlen(tax_phyloall_boldi)), " source chronograms (out of ",
        length(tax_phyloall_boldi), ") available for the ", taxon, ". To date these
        trees we are using the software PATHd8 for tree dating without a molecular
        clock model, using calibrations from all other source chronograms. Sometimes,
        calibrations conflict between them. To deal with conflicting calibrations, we can either
        expand them to make them agree, or we can congruify them to the topology
        of the tree to be dated.
        Results from both approaches are shown in the following two sections.\n\n")
}
cat(paragraph)
```

```{r echo = FALSE, warning=FALSE, message = FALSE, results = 'asis'}
p1 <- "### 3.2.1. Expanding calibrations\n\n"
# crossval_pathd8_exp <- use_calibrations(phy = source_chronogram_bold_tree, calibrations = all_other_calibrations,
#   dating_method = "pathd8", expand = default)
# crossval_pathd8_exp_notc <- use_calibrations(phy = source_chronogram_bold_tree_notc, calibrations = all_other_calibrations,
#   dating_method = "pathd8", expand = default)
# p2 <-
paragraph2 <- paste0("### 3.2.2. Summarizing calibrations (congruifying calibrations)\n\n
")
# crossval_pathd8_summ <- use_calibrations(phy = source_chronogram_bold_tree, calibrations = all_other_calibrations,
#   dating_method = "pathd8", expand = 0)
# crossval_pathd8_summ_notc <- use_calibrations(phy = source_chronogram_bold_tree_notc, calibrations = all_other_calibrations,
#   dating_method = "pathd8", expand = 0)

if(length(bold_has_brlen(tax_phyloall_boldi)) > 0){
    cat(p1)
    cat(paragraph2)
}
```




<!--
## 4. Simulate data/ Add missing taxa
Finally, if data can be added to An alternative to generate a dated tree from a set of taxa is to take the available
information and simulate into it the missing data.
We will take the median and sdm summary chronograms to date the Synthetic tree of Life:
```{r LTTplot-sims, echo = F, results = 'asis', fig.cap= ""}
#nothing yet
```
-->

\newpage

## 4. Example with subspecies tree
As an example, we're gonna date the subspecies tree of the group using all approaches
for generating new data.
```{r echo = FALSE}
# txtA <- toupper("Something is missing.")
# if(!inherits(tax_allcal_datedotol, "list")){
#   txtA <- paste(taxon, "and no tree could be constructed.")
# } else {
#     if(is.null(tax_allcal_datedotol$phy)){
#       txtA <- paste(taxon, "and they do not agree, so no tree could be constructed.")
#     } else {
#       if(all(is.na(tax_allcal_datedotol$phy$edge.length))){
#         txtA <- paste(taxon, "and a tree was constructed, but all branch lengths are NA.")
#       } else {
#           txtA <- paste(taxon, "and a tree with", ape::Ntip(tax_allcal_datedotol$phy), "tips,",
#                         round(tax_allcal_datedotol$phy$Nnode/ape::Ntip(tax_allcal_datedotol$phy)*100),
#                         "% resolved nodes and a MRCA of",
#                         round(max(ape::branching.times(tax_allcal_datedotol$phy))),"was constructed.")
#       }
#     }
# }
```

Now, let's say you like the Open Tree of Life Taxonomy and you want to stick to
that tree. Dates from available studies were tested over the Open Tree of Life
Synthetic tree of `r txtA`
We also tried  each source chronogram independently, with the Dated OToL and with
each other, as a form of cross validation in `r table_crossval`. This is not working
perfectly yet, but we are developping new ways to use all calibrations efficiently.

\newpage

# Tables and Figures

```{r echo = FALSE, warning=FALSE, message = FALSE, results = 'asis'}
table_caption <- paste(taxon, "source chronogram studies information.")
make_table0(tax_dr, tax_summ, tax_dq, table_caption)
```

<!--
\newpage

```{r echo = FALSE}
table_cap <- "Was it successful to use each source chronogram independently as calibration (CalibN) against the Dated Open Tree of Life (dOToL) and each other (ChronoN)?"
table2 <- cbind(tax_eachcal_datedotol, do.call(cbind, tax_crossval))
dimnames(table2) <- list(paste0("Calibrations", seq(tax_crossval)), c("dOToL", paste0("Chrono", seq(tax_crossval))))
if(ncol(table2) < 7) fs <- 10
if(ncol(table2) == 7 | ncol(table2) == 8) fs <- 9
if(ncol(table2) >= 9) fs <- 7
if(ncol(table2) > 10) {
  table2ori <- table2
  table2[table2 == TRUE] <- "yes" # "&#x2611;"  # :tick:
  table2[table2 == "FALSE"] <- "no"  # "&#x2612;"
  dimnames(table2) <- list(paste0("Calib", seq(tax_crossval)),
                           c("dOToL", paste0("Chr", seq(tax_crossval))))
}
tt <- knitr::kable(table2, caption = table_cap, row.names = TRUE, format = "latex", booktabs = TRUE)
kableExtra::kable_styling(kable_input = tt, position = "left", font_size = fs)
```
-->
\newpage

```{r LTTplot-phyloall, echo = F, results = 'asis', fig.cap= ""}
figcap_lttplot_phyloall <- paste0("Lineage through time (LTT) plots of source chronograms
  available in database for species in the ", taxon, ". Numbers correspond to original
  studies in ", table_chronograms, ". Arrows indicate maximum age of each chronogram.")
lttplot_phyloall <- paste0("\n\\begin{figure}[h!]\n",
    "\t\\includegraphics{plots/", taxon, "_LTTplot_phyloall.pdf}\n",
    "\t\\caption{", figcap_lttplot_phyloall, "}\n",
    "\t\\label{fig:lttplot_phyloall}\n",
    "\\end{figure}")
cat(lttplot_phyloall)
```

\begin{figure}[!h]
\includegraphics{`r paste0("plots/", taxon, "_LTTplot_phyloall.pdf")`}
\caption{`r figcap_lttplot_phyloall`}
\label{fig:lttplot_phyloall2}
\end{figure}




\newpage

```{r LTTplot-cluster-median, echo = F, results = 'asis', fig.cap= ""}
figcap_lttplot <- paste0("Lineage Through Time plots of ", taxon, " median summary
chronograms obtained with different clustering algorithms. Not all algorithms worked
with this summary matrix and we are only showing here the ones that worked.
Chronograms obtained from the SDM summary matrix are very similar to the ones from
the median summary matrix with all clustering algorithms (Appendix Fig. \\ref{fig:lttplot_cluster_sdm}).")

lttplot_cluster <- paste0("\n\\begin{figure}[h!]\n",
    "\t\\includegraphics{plots/", taxon, "_lttplot_cluster_median.pdf}\n",
    "\t\\caption{", figcap_lttplot, "}\n",
    "\t\\label{fig:lttplot_cluster_median}\n",
    "\\end{figure}")
cat(lttplot_cluster)
```

\newpage

```{r LTTplot-summ-chronograms, echo = F, results = 'asis', fig.cap= ""}
figcap_lttplot <- paste(taxon, "lineage through time (LTT) plots from
    source chronograms (gray), median (green) and SDM (blue) summary chronograms
    obtained by calibrating a consensus tree tropology with distance data
    from respective summary matrices and then adjusting branch lengths with BLADJ.")

lttplot_summchrono <- paste0("\n\\begin{figure}[h!]\n",
    "\t\\includegraphics{plots/", taxon, "_LTTplot_summary_chronograms.pdf}\n",
    "\t\\caption{", figcap_lttplot, "}\n",
    "\t\\label{fig:lttplot_summchrono}\n",
    "\\end{figure}")
cat(lttplot_summchrono)
```

\newpage

```{r LTTplot-bladj, echo = F, results = 'asis', fig.cap= ""}
figcap_lttplot <- paste(taxon, "lineage through time (LTT) plots from
    source chronograms used as secondary calibrations (gray), source chronograms
    used as topology (purple) and chronograms resulting from calibrating the latter
    with the former, using BLADJ (green).")

lttplot <- paste0("\n\\begin{figure}[h!]\n",
    "\t\\includegraphics{plots/", taxon, "_LTTplot_crossval_bladj.pdf}\n",
    "\t\\caption{", figcap_lttplot, "}\n",
    "\t\\label{fig:lttplot_crossval_bladj}\n",
    "\\end{figure}")
cat(lttplot)
```

\newpage

```{r LTTplot-pathd8-exp1, echo = F, results = 'asis', fig.cap= ""}
figcap_lttplot <- paste(taxon, "lineage through time (LTT) plots from
    source chronograms used as secondary calibrations (gray), source chronograms
    used as topology (purple) and chronograms resulting from calibrating the latter
    with the former, using PATHd8 (green).")

lttplot <- paste0("\n\\begin{figure}[h!]\n",
    "\t\\includegraphics{plots/", taxon, "_LTTplot_crossval_pathd8_exp1.pdf}\n",
    "\t\\caption{", figcap_lttplot, "}\n",
    "\t\\label{fig:lttplot_crossval_pathd8_exp1}\n",
    "\\end{figure}")
cat(lttplot)
```

\break
\newpage













# Appendix
```{r echo = FALSE, results = 'asis'}
absent <- levels(tax_median_phylo$absent_taxa$taxon)
if(!"None" %in% absent){
  cat("The following species were not found in the chronogram database: ", paste0("*", levels(tax_median_phylo$absent_taxa$taxon), collapse = "*, *", "*"))
}
```

```{r LTTplot-cluster-sdm, echo = F, results = 'asis', fig.cap= ""}
figcap_lttplot <- paste0("Lineage Through Time plots of ", taxon, " SDM summary
chronograms obtained with different clustering algorithms. Not all algorithms worked
with the SDM summary matrix and we are only showing here the ones that worked.
Chronograms obtained from the median summary matrix are very similar to the ones shown
here with all algorithms (main figure \\ref{fig:lttplot_cluster_median}).")

lttplot_cluster <- paste0("\n\\begin{figure}[h!]\n",
    "\t\\includegraphics{plots/", taxon, "_lttplot_cluster_sdm.pdf}\n",
    "\t\\caption{", figcap_lttplot, "}\n",
    "\t\\label{fig:lttplot_cluster-sdm}\n",
    "\\end{figure}")
cat(lttplot_cluster)
```


```{r myfile-1-plot, echo = F, results = 'asis', fig.cap= ""}
# knitr::include_graphics(paste0("docs/datedotol_", taxon, ".pdf"))
figcap <- paste(taxon, "Species Dated Open Tree of Life Induced Subtree. This chronogram was obtained with `get_dated_otol_induced_subtree()` function.")

if(!inherits(tax_datedotol, "phylo")){
  plot1 <- paste0("Dated induced subtree could not be obtained for the ", taxon, ".\n")
} else {
  plot1 <- paste0("\n![", figcap, "](plots/", taxon, "_datedotol.pdf)\n")
}
cat(plot1)
```


```{r eval = TRUE, echo = F, results = 'asis'}
figcap_lttplot_summ <- rep(list(""), 2)
for (i in seq_along(negs_list)){
  if(length(negs_list[[i]])>0){
    xx <- rownames(tax_summ_matrix[[i]])[ceiling(negs_list[[i]]/nrow(tax_summ_matrix[[i]]))]
    figcapextra <- "As you can note, dashed lines and solid lines from trees coming out from both types of clustering algorithms implemented are mostly overlapping. This means that removing negative values does not change results from clustering algorithms much."
    cat(paste0("This taxon's ", names(negs_list)[i], " matrix has some negative values in the following taxa: *", paste(xx, collapse = "*, *"), "*. "))
  } else {
    figcapextra <- ""
    cat(paste("This taxon's",  names(negs_list)[i], "matrix has NO negative values."))
  }
  figcap_lttplot_summ[[i]] <- paste(taxon, "lineage through time (LTT) plots from source chronograms and",
  names(negs_list)[i], "summary matrix converted to phylo with different methods (NJ and UPGMA).",
  figcapextra, "Clustering algorithms used often are returning non-ultrametric trees or
  with maximum ages that are just off (too old or too young). So we developped an
  alternative algorithm in `datelife` to go from a summary matrix to a fully ultrametric tree.")
}
```
```{r LTTplot-median, echo = F, results = 'asis', fig.cap= ""}
figure <- paste0("\n![", figcap_lttplot_summ[[2]], "](plots/", taxon, "_LTTplot_Median.pdf)\n")
cat(figure)
```

```{r LTTplot-sdm, echo = F, results = 'asis', fig.cap= ""}
lttplot_sdm <- paste0("\n![", figcap_lttplot_summ[[1]], "](plots/", taxon, "_LTTplot_sdm.pdf)\n")
cat(lttplot_sdm)
```

```{r LTTplot-median-and-sdm-summtrees, echo = F, results = 'asis', fig.cap= ""}
figcap_lttplot_summ <- rep(list(""), 2)
for(i in 1:2){
  figcap_lttplot_summ[[i]] <- paste(taxon, "lineage through time (LTT) plots from source chronograms and",
  names(negs_list)[i], "summary matrix converted to phylo with `datelife` algorithm.")
}

lttplot_median <- paste0("\n![", figcap_lttplot_summ[[2]], "](plots/", taxon, "_LTTplot_summtrees_Median.pdf)\n")
lttplot_sdm <- paste0("\n![", figcap_lttplot_summ[[1]], "](plots/", taxon, "_LTTplot_summtrees_SDM.pdf)\n")
cat(lttplot_median)
cat(lttplot_sdm)
```
